common/src/commonMain/kotlin/dev/johnoreilly/common/data/repository/FantasyPremierLeagueRepository.kt

```diff
@@ -120,46 +120,42 @@ class FantasyPremierLeagueRepository : KoinComponent {
-                                localKickoffTime,
-                                homeTeamName,
-                                awayTeamName,
-                                homeTeamPhotoUrl,
-                                awayTeamPhotoUrl,
-                                homeTeamScore,
-                                awayTeamScore,
-                                it.event
-                            )
-                            //Build gameweek to fixture map
-                            var currentValueForGw = _gameweekToFixtureMap.value[gf.event]
-                            if (currentValueForGw.isNullOrEmpty()) {
-                                _gameweekToFixtureMap.value[gf.event!!] = mutableListOf(gf)
-                            } else {
-                                currentValueForGw = currentValueForGw.plus(gf)
-                                _gameweekToFixtureMap.value[gf.event!!] = currentValueForGw
+                        _fixtureList.value = it.toList().mapNotNull {
+                            val homeTeamName = it.homeTeam?.name ?: ""
+                            val homeTeamCode = it.homeTeam?.code ?: 0
+                            val homeTeamScore = it.homeTeamScore ?: 0
+                            val homeTeamPhotoUrl =
+                                "https://resources.premierleague.com/premierleague/badges/t${homeTeamCode}.png"
+
+                            val awayTeamName = it.awayTeam?.name ?: ""
+                            val awayTeamCode = it.awayTeam?.code ?: 0
+                            val awayTeamScore = it.awayTeamScore ?: 0
+                            val awayTeamPhotoUrl =
+                                "https://resources.premierleague.com/premierleague/badges/t${awayTeamCode}.png"
+
+                            it.kickoffTime?.let { kickoffTime ->
+                                val localKickoffTime = kickoffTime.toInstant()
+                                    .toLocalDateTime(TimeZone.currentSystemDefault())
+
+                                val gf = GameFixture(
+                                    it.id,
+                                    localKickoffTime,
+                                    homeTeamName,
+                                    awayTeamName,
+                                    homeTeamPhotoUrl,
+                                    awayTeamPhotoUrl,
+                                    homeTeamScore,
+                                    awayTeamScore,
+                                    it.event
+                                )
+
+                                return@let gf
                             }
-
-                            return@let gf
                         }
-                    }
+                        //Build gameweek to fixture map
+                        _gameweekToFixtureMap.value = _fixtureList.value
+                            .filter { it.event != null }
+                            .groupBy { it.event!! }
```

Comment:
Must be cleaner way to do this with needed "!!"

Reply:
Looking at this it seems none of the fixtures in `fixtureList` have event == null. Presumably they all get lost where you check that they have a `kickOffTime`. 

So in that case we could remove the nullability from event on the `GameFixture `object which then removes that filter & double-bang ?

Though I'm not sure what those first few results from the fixtures api call are that have no kickoff time or event.

Reply:
[This MR](https://github.com/joreilly/FantasyPremierLeague/pull/44) does this.

Reply:
They have be fixtures that were cancelled/postponed but haven't been rescheduled yet